bin/faucet/src/server/resources/index.js

```diff
@@ -84,6 +113,52 @@ document.addEventListener('DOMContentLoaded', function () {
         publicButton.disabled = false;
     }
 
+    // Function to find a valid nonce for proof of work
+    async function findValidNonce(seed, difficulty) {
```

Comment:
Similar to comment above about the hex string - would we consider using an existing crate like this https://docs.rs/pow/latest/pow/

It has a helper function to calculate expected duration based on difficulty which might help.

I haven't used that crate or looked into it. Just wondering.

Reply:
I'm not quite sure that I'm fully understanding this comment, you meant in the server side, right?

Reply:
Ah yea sorry I had too brief a look. I'll find some quiet time later today and look again.

In general I'm interested in how we would unit test the logic given the frontend is performing the pow algorithm.

Reply:
Yea I guess if there isn't a js library to replace this pow logic with we just need to unit test this function?

Reply:
Not sure how to add a unit test for this function

Reply:
In terms of the js project or the function specifically? I'm pretty dusty with js so if no one else has any ideas we can come back to this. Think its worth having a unit test for this b/c of the impact it could have if faulty.

Reply:
I saw that we need to integrate a test framework to test this. I think the best will be to create end-to-end tests for the faucet, created: https://github.com/0xMiden/miden-node/issues/848